assets/javascripts/discourse/lib/activity-pub-utilities.js

```diff
@@ -7,3 +11,140 @@ export function buildHandle({ actor, model, site }) {
+  );
+}
+
+export function activityPubPostStatus(post) {
+  let status;
+
+  if (post.activity_pub_deleted_at) {
+    status = "deleted";
+  } else if (post.activity_pub_updated_at) {
+    status = "updated";
+  } else if (post.activity_pub_published_at) {
+    status = post.activity_pub_local ? "published" : "published_remote";
+  } else if (post.activity_pub_scheduled_at) {
+    status = moment().isAfter(moment(post.activity_pub_scheduled_at))
+      ? "scheduled_past"
+      : "scheduled";
+  } else {
+    status = "not_published";
+  }
+
+  return status;
+}
+
+export function activityPubPostStatusText(post, opts = {}) {
+  const status = activityPubPostStatus(post);
+
+  let i18nKey;
+  let i18nOpts = {};
+
+  if (opts.showObjectType && post.activity_pub_object_type) {
+    i18nKey = "object_status";
+    i18nOpts.object_type = post.activity_pub_object_type;
+  } else {
+    i18nKey = "status";
+  }
+
+  let time;
+  if (status === "deleted") {
+    time = moment(post.activity_pub_deleted_at);
+  } else if (status === "updated") {
+    time = moment(post.activity_pub_updated_at);
+  } else if (status === "published") {
+    time = moment(post.activity_pub_published_at);
+  } else if (status === "published_remote") {
+    time = moment(post.activity_pub_published_at);
+    i18nOpts.domain = post.activity_pub_domain;
+  } else if (status.includes("scheduled")) {
+    time = moment(post.activity_pub_scheduled_at);
+  }
+
+  if (time) {
+    i18nOpts.time = time.format("h:mm a, MMM D");
+  }
+
+  return i18n(`post.discourse_activity_pub.${i18nKey}.${status}`, i18nOpts);
+}
+
+export function activityPubTopicStatus(topic) {
+  let status;
+
+  if (topic.activity_pub_deleted_at) {
+    status = "deleted";
+  } else if (topic.activity_pub_published_at) {
+    status = topic.activity_pub_local ? "published" : "published_remote";
+  } else if (topic.activity_pub_scheduled_at) {
+    status = moment().isAfter(moment(topic.activity_pub_scheduled_at))
+      ? "scheduled_past"
+      : "scheduled";
+  } else {
+    status = "not_published";
+  }
+
+  return status;
+}
+
+export function activityPubTopicStatusText(topic, opts = {}) {
+  const status = activityPubTopicStatus(topic);
+
+  let i18nKey = "status";
+  let i18nOpts = {};
+
+  if (opts.showObjectType) {
+    i18nKey = "object_status";
+    i18nOpts.object_type = topic.activity_pub_object_type || "Collection";
+  }
+
+  let time;
+  if (status === "deleted") {
+    time = moment(topic.activity_pub_deleted_at);
+  } else if (status === "published") {
+    time = moment(topic.activity_pub_published_at);
+  } else if (status === "published_remote") {
+    time = moment(topic.activity_pub_published_at);
+    i18nOpts.actor = topic.activity_pub_actor.handle;
+  } else if (status.includes("scheduled")) {
+    time = moment(topic.activity_pub_scheduled_at);
+  }
+
+  if (time) {
+    i18nOpts.time = time.format("h:mm a, MMM D");
```

Comment:
I think we need the year here. In fact, to improve localization, maybe we can use `long_with_year` here?

Reply:
Done.